src/main/java/step2/stringcalculator/StringCarculator.java

```diff
@@ -0,0 +1,39 @@
+package step2.stringcalculator;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class StringCarculator {
+
+    public int getStringCalculator(String inputExpression) {
+        CommonException.inputExpressionInValidException(inputExpression);
+
+        List<String> expressions = Arrays.asList(inputExpression.split(CommonConstant.DELIMITER));
+        int resultValue = Integer.parseInt(expressions.get(0));
+
+        for (int i = 1; i < expressions.size(); i += 2) {
+            String operator = expressions.get(i);
+            int number = Integer.parseInt(expressions.get(i + 1));
+            resultValue = StringCarculator.calculate(operator, resultValue, number);
+        }
+        return resultValue;
+    }
+
+
+    public static int calculate(String operator, int firstNumber, int secondNumber) {
+        if (operator.equals(CommonConstant.PLUS_SIGN)) {
+            return Operator.PLUS.getSolve(firstNumber, secondNumber);
+        }
+        if (operator.equals(CommonConstant.MINUS_SIGN)) {
+            return Operator.MINUS.getSolve(firstNumber, secondNumber);
+        }
+        if (operator.equals(CommonConstant.MULTIPLICATION_SIGN)) {
+            return Operator.MULTIPLY_BY.getSolve(firstNumber, secondNumber);
+        }
+        if (operator.equals(CommonConstant.DIVISION_SIGN)) {
+            return Operator.DIVIDED_BY.getSolve(firstNumber, secondNumber);
+        }
```

Comment:
현재 if문 반복을 통해 연산이라는 행위가 중복으로 구현되어 있습니다!
현재 구현하신 enum에 연산자에 따른 값과 행위가 잘 명시 되어 있기에
StringCarculator가 Operator에게 연산 해달라고 메시지를 보내도록 구현한다면
if문의 중복이 제거될 수 있을 것 같아요 :)

Reply:
피드백 감사합니다.
StringCarculator가 Oeprator에게 연산 메세지를 보낸다는 말이 잘 와닿지 않은데 조금 더 조언을 부탁드려도 될까요 ??

Reply:
절차지향은 정보를 얻은후 결정하지만, 객체지향은 객체에게 그것을 하도록 시킵니다.
메시지라는 것은 독립적인 객체들이 협력하기 위한 의사소통 수단이라고 할 수 있습니다.
메시지를 수신했을때 실제로 수행되는 함수를 메서드라 할 수 있는데요.

연산을 해달라는 오퍼레이션을 StringCarculator or Operator가 수행을 해야하는데
StringCarculator보단 Operator가 수행하는게 맞다고 생각하기에 Oeprator에게 연산해달라는 메세지를 보내면 어떨지에 대해 리뷰를 드렸습니다.

현재 연산하는 행위 자체가 중복되는 이유는 StringCarculator 어떤 연산을 수행해야 될지 모르기에 if문이 중복되고 있다고 생각합니다. 어떤 연산을 수행해야 될지 모른다는건 연산에 대한 정보가 없다는 뜻으로 해석할 수 있습니다. 필요한 정보를 가장 잘 알고 있는 객체에게 메시지를 수신하도록 책임을 할당하고
그런 책임을 가지고 있는 객체들이 협력을 하는 것이 객체지향의 패러다임으로써
연산자에 대한 정보와 행위를 가지고 있는 Oeprator가 직접 연산을 수행해 결과만 반환해주면 어떨까요 :) ?

```suggestion
        Operator operator = Operator.findBy(inputOperator);
        return operator.getSolve(firstNumber, secondNumber);
```

Reply:
조언 감사드립니다.
어떤 의미인지 이해 했어요 !
다음 step3에 반영이 되겠지만
위에 operate 연산하는 메소드를 Operator 클래스로 위치를 옮겼습니다.
그리고 StringCalculator 클래스에서는 첨언해주신 코드와 같이
Operator.operate의 형식으로 메세지를 보내는 형태로 리팩토링을 했어요.

제가 아직 람다와 스트림에 익숙하지 않아 공부를 하는 중이라 if문으로 구현을 했어요.
if문 중복되는 로직을 더 줄이고 싶은데, 현재 상태는 연산 메소드의 위치만 Operator 클래스로 옮긴 형태라서 리팩토링이 크게 되었다는 생각은 들지 않더군요.

좀 더 보완할 방법이 떠오르긴 해서 업무시간에 조금 더 생각을 해보고.. step3하며 차근차근 반영하면서 시도 해보도록 해보겠습니다 .. !

Reply:
제가 말씀드린 내용은 정답이 아니지만 전달드리고자 하는 방향을 한결님께서 잘 이해해주신 것 같아요 :) 👍